### ISSUE TITLE
FITS reading converts integers to float64 due to BLANK as well as BZERO/BSCALE

### ISSUE THREAD

**Issue by TallJimbo** [astropy/astropy#7789]
The logic here:
https://github.com/astropy/astropy/blame/2192dddd7e6df3a53c824f0baefd3e249d2908f6/astropy/io/fits/hdu/image.py#L712
transforms an integer image on disk into a `float64` image in-memory if the `BLANK` keyword is set.

That's not entirely unreasonable, because a vanilla integer `numpy.ndarray` can't represent missing pixels, while a floating-point one can (via NaN).  But it's definitely surprising behavior, especially given that it's not mentioned in this FAQ:
http://docs.astropy.org/en/stable/io/fits/appendix/faq.html#why-is-an-image-containing-integer-data-being-converted-unexpectedly-to-floats

My opinion is that it would be better to respect the on-disk type and force the user to handle BLANK; other options might be to use something other than vanilla `numpy.ndarray` for `data` in this case, or even to raise an exception if the user doesn't provide a policy kwarg declaring what should be done.  If the current behavior is retained, the FAQ should definitely mention it.

**Comment by saimn** [comment]
I agree that this can be surprising and should at least be documented. One option would be to return a masked array, but it is difficult to change this now, in terms of backward compatibility.

For reference the conversion happens here:
https://github.com/astropy/astropy/blob/57de423b74cdf596754ab8ad09890d4a207a25e3/astropy/io/fits/hdu/image.py#L768-L769
